app/src/main/java/com/example/playlistmaker/presentation/SongActivity.kt

```diff
@@ -1,44 +1,46 @@
-package com.example.playlistmaker.Activity
+package com.example.playlistmaker.presentation
 
-import android.media.MediaPlayer
-import androidx.appcompat.app.AppCompatActivity
 import android.os.Bundle
-import android.os.Handler
-import android.os.Looper
 import android.view.View
 import android.widget.ImageView
 import android.widget.TextView
+import androidx.appcompat.app.AppCompatActivity
 import com.bumptech.glide.Glide
 import com.bumptech.glide.load.resource.bitmap.RoundedCorners
 import com.example.playlistmaker.R
+import com.example.playlistmaker.domain.Creator
+import com.example.playlistmaker.domain.MediaUseCase
 import java.text.SimpleDateFormat
 import java.util.*
 
 class SongActivity : AppCompatActivity() {
     private lateinit var back: ImageView
-    private lateinit var playButton:ImageView
-    private lateinit var progressOfTheWork:TextView
-    private var mediaPlayer = MediaPlayer()
-    private var handler = Handler(Looper.getMainLooper())
-    private var updateTimeRunnable: Runnable = Runnable { }
+    private lateinit var playButton: ImageView
+    private lateinit var progressOfTheWork: TextView
+    private lateinit var mediaUseCase: MediaUseCase
+
     override fun onCreate(savedInstanceState: Bundle?) {
         super.onCreate(savedInstanceState)
         setContentView(R.layout.activity_song)
+
         back = findViewById(R.id.back)
         playButton = findViewById(R.id.play_button)
+        progressOfTheWork = findViewById(R.id.progress_of_the_work)
+
         back.setOnClickListener {
             finish()
         }
-        progressOfTheWork = findViewById(R.id.progress_of_the_work)
+
+        mediaUseCase = Creator.provideMediaUseCase()
+
         val track = intent.getStringExtra("trackName")
```

Comment:
🍏 Чтобы не тянуть по одному полю из интента, наследуй класс Track от Parcelable и используй аннотацию @Parcelize. Parcelable объекты можно передавать в составе бандлов.
```suggestion
        val track = intent.getParcelableExtra<Track>("track")
```
https://developermemos.com/posts/parcelize-kotlin
https://developer.android.com/reference/kotlin/android/os/Parcelable
https://developer.android.com/kotlin/parcelize